### ISSUE TITLE
request@2.40.x has another vulnerability

### ISSUE THREAD

**Issue by stribDaveGustafson** [pkgcloud/pkgcloud#573]
nsp check:
```
Remote Memory Exposure
Name: request
Installed: 2.40.0
Vulnerable: >=2.2.6 <2.47.0 || >2.51.0 <=2.67.0
Patched: >= 2.68.0
```
https://nodesecurity.io/advisories/309

Upgrading the request package to >=2.70.0 should also resolve this:
https://github.com/pkgcloud/pkgcloud/issues/542

**Comment by GUL-** [comment]
Any info on this?

**Comment by shenghu** [comment]
request@2.40.x depends on hawk@1.1.1 which has another vulnerability. 

NSP-77

Severity: Medium
CVSS Score: 5.3

Specifically crafted long headers or uris can cause a minor denial of service when using hawk versions less than 4.1.1. "The Regular expression Denial of Service (ReDoS) is a Denial of Service attack, that exploits the fact that most Regular Expression implementations may reach extreme situations that cause them to work very slowly (exponentially related to input size). An attacker can then cause a program using a Regular Expression to enter these extreme situations and then hang for a very long time." Updates: - Updated to include fix in 3.1.3

NSP - Advisory 77: Regular Expression Denial of Service
Vulnerable Software & Versions:

hawk:< 3.1.3 || >= 4.0.0 <4.1.1

Solution: https://nodesecurity.io/advisories/77

**Comment by shenghu** [comment]
@rosskukulinski @indexzero can you please look at this issue? I tried to bump request to 2.81.x but there are a 12 more faulure. So I think it need some effort to figure out what is broken after the change.

**Comment by shenghu** [comment]
Well. this is dup w/ #531

**Comment by mattgrande** [comment]
Fixed by #587

**Comment by mattgrande** [comment]
Fixed in version 1.5.0; This can be closed.
